src/smt/theory_str_noodler/regex.h

```diff
@@ -148,6 +143,21 @@ namespace smt::noodler::regex {
      * @return sum of loops inside @p regex, with nested loops multiplied 
      */
     unsigned get_loop_sum(const app* reg, const seq_util& m_util_s);
+
+    class regex_model_fail : public default_exception {
+    public:
+        regex_model_fail() : default_exception("Failed to find model of a regex") {}
+    };
+
+    /**
+     * @brief Get some word accepted by @p regex (assumes that regex accepts something)
```

Comment:
Maybe emphasise that it is a incomplete heuristic?

Reply:
I added some comments